### PR TITLE
Add method to create a fat AABB given a margin width.

### PR DESCRIPTION
**Pull request by Rhuagh**

### CODE REVIEW

File: src/aabb.rs [hunk 1]
```diff
@@ -141,6 +141,10 @@ pub trait Aabb: Sized {
         Aabb::new(self.min() + v, self.max() + v)
     }
 
+    /// Add a margin of the given width around the AABB, returning a new AABB.
+    #[inline]
+    fn add_margin(&self, margin: Self::Scalar) -> Self;
```
Comment (kvark) [inline src/aabb.rs]: we could take `Self::Vector` and implement it in the trait?

Comment (Rhuagh) [reply src/aabb.rs]: That's possible yes. I don't mind either way.

Comment (kvark) [reply src/aabb.rs]: I'd prefer it to be a vector, assuming there are ways to turn scalar into a vector easily, if needed for the old API to be convenient

Comment (Rhuagh) [reply src/aabb.rs]: There is, yes. Array trait in cgmath.

Comment (Rhuagh) [reply src/aabb.rs]: I'll update it tonight

**Comment by Rhuagh** [comment]
Ok, I made it a vector. I've kept the implementations in the impls because right now EuclideanSpace does not allow Point - Vector, which means I either need to pollute the trait bounds (a lot), or do Self::Point::from_vec() twice. There is a PR up for adding this to cgmath, but it's not gone through yet.

**Pull request merged**